src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -140,6 +141,51 @@ func parseParams(r *http.Request) (models.RequestParams, *xhttp.ParseError) {
 	return params, nil
 }
 
+// parseInstantaneousParams parses all params from the GET request
+func parseInstantaneousParams(r *http.Request) (models.RequestParams, *xhttp.ParseError) {
```

Comment:
Added a test for this; for the endpoint there's not a lot of low hanging fruit, will put together a thin integration suite soon, or otherwise rely on Ben's work to be able to slot in fake storages

Reply:
Fair point; once we have it, something with a mock storage and `httptest` would probably more than cover it as a simple E2E test. Looks good for now!